### PR TITLE
Replaced Typebox with AJV

### PR DESCRIPTION
**Pull request by Omri-Levy**

### CODE REVIEW

File: src/util.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export type TUnionOfString<T extends string[], Acc extends TSchema[] = []> = T e
     ? TUnionOfString<R, [...Acc, TLiteral<L>]>
     : Acc;
 
-export const LiteralUnion = <const T extends string[]>(values: [...T], options?: SchemaOptions): TUnion<TUnionOfString<T>> => {
+export const LiteralUnion = <T extends string[]>(values: [...T], options?: SchemaOptions): TUnion<TUnionOfString<T>> => {
```
Comment (Omri-Levy) [inline src/util.ts:46]: Workflows service does not support Typescript v5

File: src/decorators.ts [hunk 2]
```diff
@@ -22,6 +22,8 @@ import type {
 } from './types.js';
 import { capitalize, coerceType, isObj } from './util.js';
 
+const ajv = new Ajv();
```
Comment (MatanYadaev) [inline src/decorators.ts]: Use these default values: https://github.com/fastify/ajv-compiler/blob/main/lib/default-ajv-options.js

File: src/types.ts [hunk 3]
```diff
@@ -16,68 +16,73 @@ export type MethodDecorator<T extends Function = any> = (
...
+    schema?: TTSchema;
     coerceTypes?: boolean;
     stripUnknownProps?: boolean;
     name?: string;
     required?: boolean;
     pipes?: (PipeTransform | Type<PipeTransform>)[];
 }
-export interface ResponseValidatorConfig<T extends TSchema = TSchema> extends ValidatorConfigBase {
-    schema: T;
+export interface ResponseValidatorConfig<TTSchema extends TSchema = TSchema> extends ValidatorConfigBase<TTSchema> {
```
Comment (MatanYadaev) [inline src/types.ts]: If it's easy, remove the default generic
```suggestion
export interface ResponseValidatorConfig<TTSchema extends TSchema> extends ValidatorConfigBase<TTSchema> {
```

**Review by MatanYadaev — APPROVED** [review]

File: ballerine-nestjs-typebox.tgz
Comment (MatanYadaev) [inline ballerine-nestjs-typebox.tgz:1]: I guess it shouldn't be committed. Please remove this file

**Pull request merged**